test/cpp_api_parity/torch_nn_modules.py

```diff
@@ -1 +1,143 @@
-module_metadata_map = {}
+from cpp_api_parity import TorchNNModuleMetadata
+
+# NOTE: In order to let Python/C++ API parity test pass for any of the modules here,
+# after fixing the C++ module implementation you should change change its "Implementation Parity"
+# from "No" to "Yes" in parity-tracker.md, and change the module's `TorchNNModuleMetadata` here
+# to look like:
+#
+# 'Linear': TorchNNModuleMetadata(
+#     cpp_default_constructor_args="(3, 4)",
+#     num_attrs_recursive=4,
+# )
+#
+# The meaning of each field of `TorchNNModuleMetadata` is as follows:
+#
+# `cpp_default_constructor_args`: string that represents the required non-keyword arguments
+#     for the C++ module constructor. For example, since `LinearOptions` expects two non-keyword
+#     arguments `(in_features, out_features)`, the `cpp_default_constructor_args` for `Linear`
+#     will be the string representation of any integer 2-tuple, such as "(3, 4)".
+#     Note that the C++ module constructor must take the exact same number of non-keyword arguments
+#     as the Python module constructor.
+#
+# `num_attrs_recursive`: the number of attributes (including parameters, buffers and non-tensor
+#     attributes) of a module. If the module contains any submodule, the submodule's attributes
+#     also need to be counted.
```

Comment:
Okay so from some code reading, it looks like if parity is `Yes`, then we require a `num_attrs_recursive` to be non-None. This might be a little difficult for modules like RNNs that get more weights depending on the number of layers. But I guess we can worry about that detail when we get to it.

Reply:
Sounds great - I will revise the design here when we get to the RNN modules.